samtranslator/model/api/api_generator.py

```diff
@@ -381,12 +381,8 @@ def _construct_api_domain(self, rest_api):
                     self.logical_id,
                     "HostedZoneId or HostedZoneName is required to enable Route53 support on Custom Domains.",
                 )
-            logical_id = logical_id_generator.LogicalIdGenerator(
-                "", route53.get("HostedZoneId") or route53.get("HostedZoneName")
-            ).gen()
-            record_set_group = Route53RecordSetGroup(
-                "RecordSetGroup" + logical_id, attributes=self.passthrough_resource_attributes
-            )
+            logical_id = "{}{}".format(self.logical_id, "DomainRecordSetGroup")
```

Comment:
Thanks for the PR!
What's the reason for appending `"DomainRecordSetGroup"` to `self.logical_id` instead of passing `self.logical_id` as prefix to LogicalIdGenerator constructor?

Reply:
Absolutely! I love SAM so being able to contribute back is my pleasure.

I have 2 assertions for using this format (examples below using `SomeApiGateway` as the API logical ID and `1e8f0a9f3b` as a generic hash):

a.) This follows the similar format to [ApiGatewayDeployment](https://github.com/aws/serverless-application-model/blob/58919bbf05c68183d3f76fcc8a01f4998a4512e5/samtranslator/model/api/api_generator.py#L230) or ApiGatewayBasePathMapping (as well as several others) formatting resulting in `SomeApiGatewayDeployment1e8f0a9f3b`
b.) When reviewing stack events, a logical ID of `SomeApiGateway` and `SomeApiGateway1e8f0a9f3b` does not intuitively distinguish the 2 different resources and requires cross-referencing when debugging/troubleshooting.

Reply:
After discussing with the team, we found that changing the Logical ID will force redeployments of existing RecordSetGroup resources which might break references to those ones already deployed.
One solution is to keep the logical ID for the first RecordSetGroup resource as the current implementation and update subsequent ones.

Reply:
That is a fair point that it could break existing deployments and appreciate the desire to mitigate that risk. This is an interesting idea to alter the logical ID generation based on count-of-occurrence of `HostedZoneId` or `HostedZoneName`. Perhaps the team could provide some input/suggestions on the proposed method of achieving this before going through the effort of engineering the solution:

- Inspection of the calling object. This would break Python encapsulation and could add significant fragility.
- Pass a handle to the calling object as an argument to all Generator classes enabling inspection of the generated resources to allow the class method `to_cloudformation` to handle merging of duplicate logical IDs within the class instance.
- Create a class variable as a `dict` or `list` of previously used `HostedZoneId` and `HostedZoneName`. As a dict, other meta could be shared across class instances. This has a potential negative consequence of unpredictability in tests but nothing that some regex couldn't handle.

I'm sure I'm missing other plausible solutions. :) I'm happy to implement any proposed solution but would like some input as any one of the above seems appear to add some tech-debt, in my opinion.